### PR TITLE
dbus: add context to open functions

### PR DESCRIPTION
**Pull request by vaspahomov**
This patch allows passing context value to dbus connection opening functions.

### CODE REVIEW

File: dbus/dbus.go [hunk 1]
```diff
@@ -123,27 +124,42 @@ func New() (*Conn, error) {
...
 func NewSystemdConnection() (*Conn, error) {
+	return NewSystemdConnectionContext(context.Background())
+}
+
+// NewSystemdConnectionContext same as NewSystemdConnection with context
+func NewSystemdConnectionContext(ctx context.Context) (*Conn, error) {
 	return NewConnection(func() (*dbus.Conn, error) {
 		// We skip Hello when talking directly to systemd.
-		return dbusAuthConnection(func(opts ...dbus.ConnOption) (*dbus.Conn, error) {
+		return dbusAuthConnection(ctx, func(opts ...dbus.ConnOption) (*dbus.Conn, error) {
```
Comment (lucab) [inline dbus/dbus.go:162]: I have a feeling that the context is not actually used by the `Dial` below, right?

**Comment by lucab** [comment]
Thanks for the PR! I have a doubt about context wiring, I've left a comment in review.

**Comment by lucab** [comment]
I'd be happy to see a "Deprecated" label on all the old methods, see https://github.com/golang/go/wiki/Deprecated.

File: dbus/dbus.go [hunk 2]
```diff
@@ -112,39 +113,63 @@ type Conn struct {
 
 // New establishes a connection to any available bus and authenticates.
 // Callers should call Close() when done with the connection.
+// Deprecated: use New instead
```
Comment (lucab) [inline dbus/dbus.go:116]: s/New/NewWithContext/

Comment (vaspahomov) [reply dbus/dbus.go]: done

**Comment by lucab** [comment]
@vaspahomov thanks, this looks fine to merge. Can you please squash your commits into a single one, and update the title as I did for the PR?

**Comment by vaspahomov** [comment]
@lucab done

**Review by lucab — APPROVED** [review]

**Pull request merged**